src/doc-printer.js

```diff
@@ -176,6 +194,51 @@ function printDocToString(doc, width, newLine) {
               break;
           }
           break;
+        case "fill":
+          // shouldRemeasure = false;
+          let rem = width - pos;
+
+          const parts = doc.parts;
+          if (parts.length === 1) {
+            cmds.push([ind, MODE_BREAK, parts[0]]);
+            break;
+          }
+
+          const first = parts[0];
+          const second = parts[1];
+          const remaining = parts.slice(1);
+          const firstCmd = [ind, MODE_FLAT, first];
+          const remainingCmd = [ind, MODE_BREAK, fill(remaining)];
+
+          const firstAndSecondCmd = [ind, MODE_FLAT, concat([first, second])];
```

Comment:
The basic idea is copied from fill implementation in Walder's paper (at the end of page 14).

I believe the idea is to check that the first two elements with both fit on a single line before joining them with a space.

If they don't fit on a single line then we need to join them with a new line.